### DISCUSSION TITLE
Feature request: initialize-config from environment variables

### DISCUSSION THREAD

**Discussion by lukasmrtvy** [postalserver/postal#1476] [Feature suggestions]
It would be a great for future docker support to generate postal.yml from environmental variables

For example:
```
export web.host=postal.example.com
export web.protocol=https
export main_db.host=127.0.0.1
export main_db.user=postal

postal initialize-config
```

Docker run command:

```
docker run -d  --name postal \
-e  web.host=postal.example.com \
-e  web.protocol=https \
-e  main_db.host=127.0.0.1 \
-e  main_db.user=postal \
postal:latest

```

You can even use something like:
```
 export web.protocol=${custom.web.protocol:-https}
 export web.host=${custom.web.host:-postal.example.com}
```
to perserve backward functionality

**Comment by willpower232** [comment]
According to https://stackoverflow.com/questions/13948951/update-value-of-key-of-a-yaml-file-in-ruby-on-rails Ruby is capable of updating YAML values so it would only be a minor change to https://github.com/atech/postal/blob/master/script/generate_initial_config.rb to make it so when generating the config file.

I don't think it would be a major priority right now but an interesting idea at least.
